collectd/src/main/java/com/microsoft/applicationinsights/collectd/internal/WriterConfiguration.java

```diff
@@ -0,0 +1,157 @@
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.collectd.internal;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+import com.microsoft.applicationinsights.collectd.ConfigurationException;
+import com.microsoft.applicationinsights.internal.util.LocalStringsUtils;
+import org.collectd.api.OConfigItem;
+import org.collectd.api.OConfigValue;
+
+/**
+ * Created by yonisha on 5/5/2015.
+ *
+ * This class purpose is to parse configuration passed from CollectD framework.
+ */
+public class WriterConfiguration {
+
+    // region Consts
+
+    public static final String EXCLUDE_CONFIGURATION_KEY = "Exclude";
+    public static final String INSTRUMENTATION_KEY_CONFIGURATION_KEY = "InstrumentationKey";
+    public static final String SDK_LOGGER_CONFIGURATION_KEY = "SDKLogger";
+
+    // endregion Consts
+
+    // region Members
+
+    private static ApplicationInsightsWriterLogger logger = new ApplicationInsightsWriterLogger();
+    private Dictionary<String, PluginExclusion> pluginExclusions = new Hashtable<String, PluginExclusion>();
+    private String instrumentationKey;
+    private boolean isLoggerEnabled;
+
+    // endregion Members
+
+    /**
+     * Gets the instrumentation key.
+     * @return The instrumentation key.
+     */
+    public String getInstrumentationKey() {
+        return this.instrumentationKey;
+    }
+
+    /**
+     * Gets a value indicating whether the logger is enabled or not.
+     * @return True if the logger is enabled, false otherwise.
+     */
+    public boolean getIsLoggerEnabled() {
+        return this.isLoggerEnabled;
+    }
+
+    /**
+     * Gets a dictionary of excluded plugins and data sources.
+     * @return Dictionary of excluded plugins and data sources.
+     */
+    public Dictionary<String, PluginExclusion> getPluginExclusions() {
+        return this.pluginExclusions;
+    }
+
+    /**
+     * Builds @WriterConfiguration from the given CollectD configuration object.
+     * @param configuration The CollectD configuration.
+     * @return The configuration for Application Insights writer plugin.
+     * @throws ConfigurationException Thrown if the configuration is invalid.
+     */
+    public static WriterConfiguration buildConfiguration(OConfigItem configuration) throws ConfigurationException {
+        logger.logDebug("Parsing configuration: " + configuration);
+
+        WriterConfiguration writerConfiguration = new WriterConfiguration();
+        List<OConfigItem> children = configuration.getChildren();
+
+        for (OConfigItem child : children) {
+            String key = child.getKey();
+
+            if (key.equalsIgnoreCase(INSTRUMENTATION_KEY_CONFIGURATION_KEY)) {
```

Comment:
For improved readability and cleaner code consider handling each key in its own function, i.e.

    if (key.equalsIgnoreCase(INSTRUMENTATION_KEY_CONFIGURATION_KEY)) {
        writerConfiguration.instrumentationKey = parseInstrumentationKey(child);
    } else if (key.equalsIgnoreCase(SDK_LOGGER_CONFIGURATION_KEY)) {
        writerConfiguration.isLoggerEnabled = parseIsLoggerEnabled(child);
    } else if ...